pymodbus/server/reactive/main.py

```diff
@@ -193,44 +196,43 @@ def manipulate_response(self, response):
-                logger.info("Resetting manipulator"
-                            " after {} responses".format(clear_after))
-                self.update_manipulator_config(dict(DEFAULT_MANIPULATOR))
-                return response
-            response_type = self._manipulator_config.get("response_type")
-            if response_type == "error":
-                error_code = self._manipulator_config.get("error_code")
-                logger.warning(
-                    "Sending error response for all incoming requests")
-                err_response = ExceptionResponse(response.function_code, error_code)
-                err_response.transaction_id = response.transaction_id
-                err_response.unit_id = response.unit_id
-                response = err_response
-                self._counter += 1
-            elif response_type == "delayed":
-                delay_by = self._manipulator_config.get("delay_by")
-                logger.warning(
-                    "Delaying response by {}s for "
-                    "all incoming requests".format(delay_by))
-                time.sleep(delay_by)
-                self._counter += 1
-            elif response_type == "empty":
-                logger.warning("Sending empty response")
-                self._counter += 1
-                response.should_respond = False
-            elif response_type == "stray":
-                data_len = self._manipulator_config.get("data_len", 10)
-                if data_len <= 0:
-                    logger.warning(f"Invalid data_len {data_len}. "
-                                   f"Using default lenght 10")
-                    data_len = 10
-                response = os.urandom(data_len)
-                self._counter += 1
-                skip_encoding = True
-            return response, skip_encoding
+
+        clear_after = self._manipulator_config.get("clear_after")
+        if clear_after and self._counter > clear_after:
+            txt = f"Resetting manipulator {clear_after} responses"
+            logger.info(txt)
+            self.update_manipulator_config(dict(DEFAULT_MANIPULATOR))
+            return response
+        response_type = self._manipulator_config.get("response_type")
+        if response_type == "error":
+            error_code = self._manipulator_config.get("error_code")
+            logger.warning(
+                "Sending error response for all incoming requests")
+            err_response = ExceptionResponse(response.function_code, error_code)
+            err_response.transaction_id = response.transaction_id
+            err_response.unit_id = response.unit_id
+            response = err_response
+            self._counter += 1
+        elif response_type == "delayed":
+            delay_by = self._manipulator_config.get("delay_by")
+            txt = f"Delaying response by {delay_by}s for all incoming requests"
+            logger.warning(txt)
+            time.sleep(delay_by)
```

Comment:
Note for self : This method should ideally be `async` and we should be using . TODO

Reply:
you are quite right, I have a note of changing all time.sleep, this will however be easier when there is a decision to remove tornado and twister.

Reply:
👍 thanks did not know tat.